fix: ProductCard 타입 수정 반영 및 빌드 오류 해결

## 🐞 버그 설명

<!-- 어떤 문제가 발생했는지 명확하게 써주세요. (예: 로그인 버튼 클릭 시 아무 반응 없음) -->
`src/features/product/ProductCard.tsx(16,40): error TS2339: Property 'product_card_image' does not exist on type '{ product_card_image: string; }[]'.` 빌드 오류가 뜸

## ⚙️ 재현 방법

<!-- 1. 페이지 접속
2. 버튼 클릭
3. 오류 메시지 혹은 동작 이상 확인 -->

1. vercel에 배포

## ✅ 예상 동작

<!-- 정상적으로 작동해야 하는 상황 설명 -->
빌드가 됨...

## 🧠 원인 추정 (선택)

<!-- 대충 어디서 문제가 생겼을 것 같은지 (예: 상태 관리, 비동기 처리 등) -->
`ProductCard` 컴포넌트를 같이 커밋하지 않음

## 🔗 관련 이슈/PR

<!-- 관련된 이슈나 참고 링크 -->

--- a/src/features/product/ProductCard.tsx
+++ b/src/features/product/ProductCard.tsx
@@ -13,7 +13,7 @@ export function ProductCard({ product }: ProductCardProps) {
       <div className='flex max-h-[600px] min-h-96 min-w-60 flex-col items-center justify-center gap-4 text-center'>
         <div className='relative flex h-full w-full flex-col'>
           <img
-            src={product.product_image.product_card_image}
+            src={product.product_image[0]?.product_card_image}
             alt={product.product_name}
             className='h-auto w-full min-w-32 object-cover object-top'
             width={300}

--- a/src/types/product.ts
+++ b/src/types/product.ts
@@ -23,9 +23,11 @@ export interface ProductCardType {
   product_value: string;
   dc_value: number;
   product_rating: string;
-  product_image: {
-    product_card_image: string;
-  }[];
+  product_image: [
+    {
+      product_card_image: string;
+    },
+  ];
 }
 
 export interface ProductDetailType {

